Thank you, for contributing.

I teseted this PR on Linux, compiler(gcc) print errors as below:

```
/usr/include/stdc-predef.h:1: fatal error: can’t create precompiled header generated/tinyvm: No such file or directory
```

This error can be solved by to make a `generated` directory before compile.


I also test this makefile on macOS, but this makefile didn't work and compiler(clang, because cc/gcc is aliased to clang on macOS) print errors as below:

```
clang: error: cannot specify -o when generating multiple output files
```

I aim to be able to compile tinyvms/c_impl by clang(I wrote `gcc` to makefile, though. It is because that, as you probably know, on macOS `gcc` is alised to clang).
Then, if you can, please adapt this makefile to clang.

--- a/c_impl/.gitignore
+++ b/c_impl/.gitignore
@@ -1,0 +1,2 @@
+generated/
+*.o

--- a/c_impl/Makefile
+++ b/c_impl/Makefile
@@ -1,5 +1,20 @@
-all:
-	gcc -o tinyvm *.c sds/*.c -g -Wall
+.PHONY: all clean
+
+CC := cc
+CFLAGS := -Wextra -Wall -g
+
+TARGET = tinyvm
+SRCS = \
+	$(shell find ./ -name "*.c") \
+	$(shell find ./sds -name "*.c")
+OBJS = $(shell find ./ -name "*.o")
+
+GENERATED = generated
+
+all: $(TARGET)
+
+$(TARGET): $(SRCS)
+	$(CC) -o $(addprefix $(GENERATED)/, $@) $^ $(CFLAGS)
 
 clean:
-	rm *.o tinyvm
+	$(RM) $(OBJS) $(addprefix $(GENERATED)/, $(TARGET))

--- a/c_impl/Makefile
+++ b/c_impl/Makefile
@@ -13,8 +13,11 @@ GENERATED = generated
 
 all: $(TARGET)
 
-$(TARGET): $(SRCS)
+$(TARGET): $(SRCS) | $(GENERATED)
 	$(CC) -o $(addprefix $(GENERATED)/, $@) $^ $(CFLAGS)
+
+$(GENERATED):
+	mkdir -p $(GENERATED)
 
 clean:
 	$(RM) $(OBJS) $(addprefix $(GENERATED)/, $(TARGET))

--- a/c_impl/util.c
+++ b/c_impl/util.c
@@ -1,6 +1,7 @@
 #include "tinyvm.h"
 #include <assert.h>
 #include <stdbool.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>

